.pre-commit-config.yaml

```
@@ -81,48 +106,65 @@ repos:
       - id: shell-lint
 
   # Python hooks
+  # Run bandit on the "tests" tree with a configuration
   - repo: https://github.com/PyCQA/bandit
-    rev: 1.7.4
+    rev: 1.7.5
     hooks:
       - id: bandit
+        name: bandit (tests tree)
+        files: tests
         args:
           - --config=.bandit.yml
-  - repo: https://github.com/psf/black
-    rev: 22.3.0
+  # Run bandit on everything except the "tests" tree
+  - repo: https://github.com/PyCQA/bandit
+    rev: 1.7.5
+    hooks:
+      - id: bandit
+        name: bandit (everything else)
+        exclude: tests
+  - repo: https://github.com/psf/black-pre-commit-mirror
+    rev: 23.9.1
     hooks:
       - id: black
-  - repo: https://gitlab.com/pycqa/flake8
-    rev: 3.9.2
+  - repo: https://github.com/PyCQA/flake8
+    rev: 6.1.0
     hooks:
       - id: flake8
         additional_dependencies:
           - flake8-docstrings
   - repo: https://github.com/PyCQA/isort
-    rev: 5.10.1
+    rev: 5.12.0
     hooks:
       - id: isort
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.5.1
+    hooks:
+      - id: mypy
+        additional_dependencies:
+          - types-PyYAML
+          - types-setuptools
```

This adds typing for other packages used and adjusts the arguments being used by the hook.

```suggestion
        additional_dependencies:
          - boto3-stubs
          - pymongo-stubs
          - types-chevron
          - types-docopt
          - types-PyYAML
          - types-setuptools
        args:
          # Default arguments
          - --ignore-missing-imports
          - --scripts-are-modules
          # Support native namespace packages with pre-commit
          - --explicit-package-bases
```